lib/active_record/connection_adapters/oracle_enhanced_adapter.rb

```diff
@@ -527,7 +527,7 @@ def tables #:nodoc:
       end
 
       def data_sources
-        super
+        tables | views | synonym_names
```

Comment:
how about `super | synonym_names`?  Oracle enhanced adapter should not be affected in case of Abstract adapter `data_sources` implementation changes.